src/sql/SqlService.ts

```diff
@@ -42,6 +42,92 @@ import {HzLocalTime, HzLocalDate, HzLocalDateTime, HzOffsetDateTime} from './Dat
 
 /**
  * SQL Service of the client. You can use this service to execute SQL queries.
+ *
+ * The service is in beta state. Behavior and API might change in future releases. Binary compatibility is not
+ * guaranteed between minor or patch releases.
+ *
+ * If this cluster is a Hazelcast Jet cluster, a statement can be executed by either the default SQL backend or by Hazelcast
+ * Jet backend, as a Jet job. If some of the features used in a statement isn't supported by the default backend, the engine will
+ * attempt to execute it using Jet. This class is the API to both backends.
+ *
+ * The text below summarizes features supported by the default SQL engine. For a summary of Hazelcast Jet SQL features
+ * see `com.hazelcast.jet.sql` package javadoc in Hazelcast Jet (once released).
+ *
+ *
+ * ### Overview
+ * Hazelcast is able to execute distributed SQL queries over the following entities:
+ * * IMap
+ *
+ * When an SQL statement is submitted to a member, it is parsed and optimized by the `hazelcast-sql` module, that is based
+ * on <a href="https://calcite.apache.org">Apache Calcite</a>. The `hazelcast-sql` must be in the classpath, otherwise
+ * an exception will be thrown.
+ *
+ * During optimization, a statement is converted into a directed acyclic graph (DAG) that is sent to cluster members
+ * for execution. Results are sent back to the originating member asynchronously and returned to the user via {@link SqlResult}.
+ *
+ * ##### Querying an IMap
+ *
+ * Every IMap instance is exposed as a table with the same name in the `partitioned` schema. The `partitioned`
+ * schema is included into a default search path, therefore an IMap could be referenced in an SQL statement with or without the
+ * schema name.
+ *
+ * ###### Column Resolution
+ * Every table backed by an IMap has a set of columns that are resolved automatically. Column resolution uses IMap entries
+ * located on the member that initiates the query. The engine extracts columns from a key and a value and then merges them
+ * into a single column set. In case the key and the value have columns with the same name, the key takes precedence.
+ *
+ * Columns are extracted from objects as follows:
```

Comment:
We can note that resolution happens on the server-side. 

I used something like this in the Python doc.

```
Columns are extracted from objects as follows (which happens on the
    server-side)
```